base/applications/clipbrd/cliputils.c

```diff
@@ -191,59 +192,146 @@ BOOL IsClipboardFormatSupported(UINT uFormat)
 
         default:
         {
+            if (uFormat == Globals.uCFSTR_FILENAMEA ||
+                uFormat == Globals.uCFSTR_FILENAMEW)
+            {
+                return TRUE;
+            }
+
             return FALSE;
         }
     }
 }
 
-SIZE_T
-GetLineExtentW(
-    IN LPCWSTR lpText,
-    OUT LPCWSTR* lpNextLine)
+#define CALC_TEXT_SIZE_FORMAT (DT_CALCRECT | DT_LEFT | DT_TOP | DT_SINGLELINE | DT_NOPREFIX)
+
+SIZE GetTextSizeW(HDC hDC, LPCWSTR pszText)
 {
-    LPCWSTR ptr;
+    SIZE textSize = { 0, 0 };
+    LPCWSTR pch0 = pszText, pch1;
+    RECT rc;
+    size_t cchLine;
 
-    /* Find the next line of text (lpText is NULL-terminated) */
-    /* For newlines, focus only on '\n', not on '\r' */
-    ptr = wcschr(lpText, L'\n'); // Find the end of this line.
-    if (ptr)
-    {
-        /* We have the end of this line, go to the next line (ignore the endline in the count) */
-        *lpNextLine = ptr + 1;
-    }
-    else
+    for (;; pch0 = pch1 + 1)
     {
-        /* This line was the last one, go pointing to the terminating NULL */
-        ptr = lpText + wcslen(lpText);
-        *lpNextLine = ptr;
+        pch1 = wcschr(pch0, L'\n');
+        cchLine = (pch1 ? (pch1 - pch0) : wcslen(pch0));
+
+        SetRectEmpty(&rc);
+        textSize.cy += DrawTextW(hDC, pch0, cchLine, &rc, CALC_TEXT_SIZE_FORMAT);
```

Comment:
Since the original code was calling GetTabbedTextExtentW, I wonder whether this DrawTextW replacement call returns sizes different from the following:
```c
    dwSize = GetTabbedTextExtentW(hDC, pch0, cchLine, 0, NULL);
```
(here, `HIWORD(dwSize)` would be the equivalent of what DrawText returns, while `LOWORD(dwSize)` would be the equivalent of `rc.right - rc.left`).

Reply:
Okay, `GetTabbedTextExtentW` is used.